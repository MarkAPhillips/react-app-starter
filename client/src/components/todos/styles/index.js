import styled from 'styled-components';
import { colours } from '../../../assets/styles/variables';

export const ContainerPanel = styled.div`
  &:hover {
    cursor: ${props => (props.isAction ? 'pointer' : 'auto')};
  }
`;

export const TodoContainerPanel = styled.section`
  border-radius: 7px;
  background-color: ${colours.lightgrey};
  padding: 1em;
`;

export const TodoListItemPanel = styled.div`
  display: flex;
  padding: 0.5em;
  align-items: center;
  width: 90%;
  justify-content: space-between;
`;

export const TodoListPanel = styled.div`
  padding: 0.5em;
  background-color: ${colours.white};
  margin: 12px 0;
  border-radius: 3px;
  > div {
    border-bottom: 1px solid ${colours.lightgrey};
    margin-bottom: 8px;
    &:last-child {
      border-bottom: none;
      margin-bottom: 0;
    } 
    &:hover {
      opacity: 0.7;
    }
  }
`;

export const TodoFilterPanel = styled.div``;

export const InputPanel = styled.div`
  width: 400px;
`;
