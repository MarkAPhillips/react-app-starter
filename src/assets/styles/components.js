import styled from 'styled-components';

export const Main = styled.main`
    padding: 1em;
`;

export const Header = styled.h1`
    font-size: 1.5em;
`;

export const Section = styled.section``;

export const Button = styled.button``;

export const Input = styled.input``;
