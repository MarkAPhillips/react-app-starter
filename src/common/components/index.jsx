import React from 'react';
import styled from 'styled-components';
import Chart from '../../charts/components/Chart';

const MainPanel = styled.main``;

const App = () => <MainPanel><Chart /></MainPanel>;

export default App;
