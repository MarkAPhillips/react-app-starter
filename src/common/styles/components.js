import styled from 'styled-components';

export const Main = styled.main`
    padding: 1em;
`;

export const Header = styled.h1`
    font-size: 1.5em;
`;

export const Section = styled.section``;
