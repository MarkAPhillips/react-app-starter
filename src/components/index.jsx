import React from 'react';
import styled from 'styled-components';

const MainPanel = styled.main``;

const App = () => <MainPanel />;

export default App;
