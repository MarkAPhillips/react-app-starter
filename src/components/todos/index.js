export { TodoForm } from './TodoForm';
export { TodoList } from './TodoList';
export { TodoListItem } from './TodoListItem';
export { TodoContainer } from './TodoContainer';
