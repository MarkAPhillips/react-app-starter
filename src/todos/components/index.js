export { TodoAddItem } from './TodoAddItem';
export { TodoList } from './TodoList';
export { TodoListItem } from './TodoListItem';
