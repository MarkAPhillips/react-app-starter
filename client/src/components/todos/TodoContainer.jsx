import React, { useState } from 'react';
import { TodoContainerPanel } from './styles';
import { Subheader } from '../../assets/styles/components';
import { TodoForm, TodoList } from './';


export const TodoContainer = () => {
  const { inputValue, setInputValue } = useState('');

  const handleChange = (evt) => {
    const { value } = evt.target;
    console.log('handleChange', inputValue);
    setInputValue(value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    console.log('handleSubmit', inputValue);
    setInputValue('');
  };

  const handleStatusChange = (evt) => {
    const { checked, id } = evt.target;
    console.log('handleStatusChage', checked, id);
  };

  return (
    <TodoContainerPanel>
      <Subheader>Inbox</Subheader>
      <TodoList
        onStatusChange={handleStatusChange}
      />
      <TodoForm
        onSubmit={handleSubmit}
        onChange={handleChange}
        inputValue={inputValue}
      />
    </TodoContainerPanel>
  );
};
