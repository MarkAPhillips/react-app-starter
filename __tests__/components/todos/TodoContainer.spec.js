import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import 'jest-styled-components';
import configureMockStore from 'redux-mock-store';
import { TodoContainer } from '../../../src/components/todos';
import { componentSnapshotShouldMatch } from '../../testUtils';

describe('TodoContainer specs', () => {
  const mockStore = configureMockStore();
  const initialState = {
    todos: {
      list: {},
    },
  };
  let component;
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    component = <Provider store={store}><TodoContainer /></Provider>;
    wrapper = mount(component);
  });

  it('should render the component ', () => {
    expect(wrapper.find(TodoContainer).length).toEqual(1);
  });

  it('should render the component snapshot correctly', () => {
    componentSnapshotShouldMatch(component);
  });
});
