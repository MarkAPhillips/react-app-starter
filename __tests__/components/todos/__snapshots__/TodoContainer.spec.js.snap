// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoContainer specs should render the component snapshot correctly 1`] = `
<section
  className=""
>
  <div
    className=""
  >
    <input
      className=""
      id="input-add-todo"
      onChange={[Function]}
      type="text"
    />
    <button
      className=""
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</section>
`;
